content/blog/setting-up-harbor-as-a-local-container-registry-in-hpe-private-cloud-ai.md

```diff
@@ -0,0 +1,304 @@
+
+## Harbor
+
+*Harbor* is an open-source container registry designed for cloud-native environments like K8s. It securely stores and manages container images with policies and RBAC, ensures images are scanned and free from vulnerabilities, and signs images as trusted.  
+
+The following sections describe in detail how to deploy *Harbor* into HPE PCAI using the *Import Framework*. You will learn how to create a private project, create users and assign them with specific role permissions, and push images using *Harbor* credentials. Used as a local image registry within HPE PCAI, *Harbor* helps ensure your container images remain secure and well governed. 
+
+### Harbor deployment via HPE PCAI *Import Framework*
+
+Based on the latest Helm charts from the official [*Harbor* site](https://helm.goharbor.io/harbor-1.17.0.tgz), the following YAML manifest files have been added under *templates/ezua/* directory:
+
+* *virtualService.yaml*: Defines an Istio *VirtualService* to configure routing rules for incoming requests.
+* *kyverno-cluster-policy*: A Kyverno *ClusterPolicy* that automatically adds required labels to the deployment.
+
+Additionally, the default *values.yaml* file has been modified with the following updates:
+
+* introduced an *ezua* section to configure the *Istio Gateway* and expose the endpoint: 
+
+```bash
+      ezua:
+
+        virtualService:
+          endpoint: "harbor.${DOMAIN_NAME}"
+          istioGateway: "istio-system/ezaf-gateway"
+```
+
+* expanded *Harbor* registry storage from the default *5G* to *500G*:
+
+```bash
+      persistence.persistentVolumeClaim.registry.size = 500G
+```
+
+These updates are implemented in the revised *Harbor* Helm charts, available in the *GitHub* repository [*pcai-helm-examples*](https://github.com/GuopingJia/pcai-helm-examples/tree/main/harbor). With these customizations, *Harbor* can be easily deployed into HPE PCAI using the *Import Framework*:
+
+![](/img/import-harbor.png)
+
+### Harbor UI access via its endpoint
+
+After *Harbor* is deployed via the HPE PCAI *Import Framework*, an **Imported** *Harbor* tile appears under *Tools & Frameworks* on the *Data Science* tab. A service endpoint, e.g., *https://harbor.ingress.pcai0104.ld7.hpecolo.net*, is automatically configured and exposed, providing access to *Harbor*. 
+ 
+![](/img/harbor-deployment.png)
+
+Click the *Open* button, or paste the endpoint URL into your browser, to launch the *Harbor* login page:
+
+![](/img/harbor-login.png)
+
+From there, you can log into *Harbor* projects page using the default *admin* user credentials:
+
+![](/img/harbor-ui.png)
+
+### Harbor project and user creation
+
+*Harbor* manages container images through projects, each of which hosts the image repositories for your application. Before pushing images to *Harbor*, a project must first be created. A default public project named *library* is pre-created, but new projects can be created by clicking *+ NEW PRORJECT*:
```

Comment:
Oh my... I thought I was going to tell you at the very end "great job... no changes!" but I do believe I have caught a typo here. "NEW PROJECT" (not PROR-JECT, lol)